builtins/generic.ispc

```diff
@@ -1381,7 +1381,15 @@ EXT uniform uint64 __cast_mask_to_i64(UIntMaskType mask) { return @llvm.ispc.pac
 // mask operations
 EXT uniform int64 __movmsk(UIntMaskType mask) { return __cast_mask_to_i64(mask); }
 EXT uniform bool __any(UIntMaskType mask) { return __cast_mask_to_i64(mask) != 0; }
-EXT uniform bool __all(UIntMaskType mask) { return __cast_mask_to_i64(mask) == (1ULL << TARGET_WIDTH) - 1; }
+EXT uniform bool __all(UIntMaskType mask) {
+#if TARGET_WIDTH == 64
+    //  Shifting a 64-bit value left by 64 positions is undefined behavior
+    //  so check mask explicitely
+    return __cast_mask_to_i64(mask) == 0xFFFFFFFFFFFFFFFF;
+#else
+    return __cast_mask_to_i64(mask) == (1ULL << TARGET_WIDTH) - 1;
```

Comment:
What about this? 😎 
```cpp
~(~0ULL << N);
```

Reply:
`~(~0ULL << TARGET_WIDTH);` causes warning `Binary expression with type "const uniform unsigned int64" can't represent value.`
But `~0ULL >> (64 - TARGET_WIDTH);` works without warning.

Reply:
Heh, right, warnings is correct, but we consciously doing what we are doing.

`~0ULL >> (64 - TARGET_WIDTH);` looks good.